storage/decorator.py

```diff
@@ -236,6 +236,21 @@ def wrapper(
 
             yield line, _, _, newline, None
 
+    def _collect_single(iterator: StorageDecoratorIterator) -> StorageDecoratorIterator:
+        single: list[tuple[Union[StorageReader, bytes], int, int, bytes]] = list(
+            [
+                (data, starting_offset, ending_offset, newline)
+                for data, starting_offset, ending_offset, newline, _ in iterator
+            ]
+        )
```

Comment:
I find this DS a bit hard to digest. Maybe a comment with an example entry could help